util/ci/macos_build_deps.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash -e
+
+# macOS build only
+download_macos_deps() {
+	brew install cmake nasm wget m4
+	brew info m4
+
+	mkdir deps
+	cd deps
+
+	echo "Downloading sources..."
+	wget -O gettext.tar.gz https://ftp.gnu.org/gnu/gettext/gettext-0.22.5.tar.gz
+	wget -O freetype.tar.xz https://downloads.sourceforge.net/project/freetype/freetype2/2.13.3/freetype-2.13.3.tar.xz
+	wget -O gmp.tar.xz https://ftp.gnu.org/gnu/gmp/gmp-6.3.0.tar.xz
+	wget -O libjpeg-turbo.tar.gz https://github.com/libjpeg-turbo/libjpeg-turbo/releases/download/3.0.3/libjpeg-turbo-3.0.3.tar.gz
+	wget -O jsoncpp.tar.gz https://github.com/open-source-parsers/jsoncpp/archive/refs/tags/1.9.5.tar.gz
+	wget -O leveldb.tar.gz https://github.com/google/leveldb/archive/refs/tags/1.23.tar.gz
+	wget -O libogg.tar.gz https://ftp.osuosl.org/pub/xiph/releases/ogg/libogg-1.3.5.tar.gz
+	wget -O libpng.tar.xz https://downloads.sourceforge.net/project/libpng/libpng16/1.6.43/libpng-1.6.43.tar.xz
+	wget -O libvorbis.tar.gz https://github.com/sfence/libvorbis/archive/refs/tags/v1.3.7_macos_apple_silicon.tar.gz
+	wget -O luajit.tar.gz https://github.com/LuaJIT/LuaJIT/archive/f725e44cda8f359869bf8f92ce71787ddca45618.tar.gz
+	wget -O zstd.tar.gz https://github.com/facebook/zstd/archive/refs/tags/v1.5.6.tar.gz
+	# leveldb depends
+	wget -O gperftools.tar.gz https://github.com/gperftools/gperftools/releases/download/gperftools-2.15/gperftools-2.15.tar.gz
+	wget -O snappy.tar.gz https://github.com/google/snappy/archive/refs/tags/1.2.1.tar.gz
+
+	cd ..
+}
+
+untar_macos_deps() {
+	cd deps
+
+	echo "Unarchiving sources..."
+	tar -xf libpng.tar.xz
+	tar -xf gettext.tar.gz
+	tar -xf freetype.tar.xz
+	tar -xf gmp.tar.xz
+	tar -xf libjpeg-turbo.tar.gz
+	tar -xf jsoncpp.tar.gz
+	tar -xf leveldb.tar.gz
+	tar -xf libogg.tar.gz
+	tar -xf libvorbis.tar.gz
+	tar -xf luajit.tar.gz
+	tar -xf zstd.tar.gz
+
+	tar -xf gperftools.tar.gz
+	tar -xf snappy.tar.gz
+
+	cd ..
+}
+
+compile_macos_deps() {
+	cd deps
+	mkdir install
+
+	dir=$(pwd)
+	export MACOSX_DEPLOYMENT_TARGET=10.4
+	export CMAKE_PREFIX_PATH=$dir/install
+
+	# libpng
```

Comment:
FWIW going all the way to this extent just to support old macOS versions *with official builds* could be considered unreasonable.
maybe we could instead support the oldest version brew has bottles for?

Reply:
It will be an easier way to support MacOS 13 and above.
However, MacOS Catalina 10.5 looks to be one of the most used MacOS systems this time. So it would be nice if we could support it.

I tried 10.4 as default because the Minetest elf file is compiled for this version.